### PR TITLE
Makes census-java compatible with Java 6

### PR DESCRIPTION
**Pull request by sebright**
This PR includes #2, because it uses the Maven build files to build with Java 6 and 7 in .travis.yml.

### CODE REVIEW

File: core/javatests/com/google/census/CensusScopeTest.java [hunk 1]
```diff
@@ -41,58 +41,76 @@ public void testNoScope() {
 
   @Test
   public void testScope() {
-    try (CensusScope scope = new CensusScope(Census.getCurrent().with(K1, V1))) {
+    CensusScope scope = new CensusScope(Census.getCurrent().with(K1, V1));
```
Comment (adriancole) [inline core/javatests/com/google/census/CensusScopeTest.java]: not sure if bazel does this, but not really required to limit tests to java6 is it?

File: core_native/java/com/google/census/CensusSerializer.java [hunk 2]
```diff
@@ -36,14 +34,23 @@ static ByteBuffer serialize(CensusContextImpl context) {
           .append(TAG_DELIM)
           .append(tag.getValue());
     }
-    return ByteBuffer.wrap(builder.toString().getBytes(UTF_8));
+    try {
+      return ByteBuffer.wrap(builder.toString().getBytes("UTF-8"));
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError("UTF-8 not supported.");
```
Comment (adriancole) [inline core_native/java/com/google/census/CensusSerializer.java]: worthwhile making this a static constant.

```
  public static final Charset UTF_8 = Charset.forName("UTF-8");
```

File: core/javatests/com/google/census/CensusScopeTest.java (hunk 1 above)
Comment (sebright) [inline core/javatests/com/google/census/CensusScopeTest.java]: If Census supports Java 6, then I think we should also run the tests with it.  I'd like to add the tests to pom.xml soon.

**Comment by sebright** [comment]
I rebased and updated the Maven build files to work with the Protocol Buffers that were introduced in #8.  #8 also removed the use of `StandardCharsets.UTF_8`.

**Pull request merged**